src/core/raster/qgsrasterblock.cpp

```diff
@@ -872,3 +872,293 @@ QRect QgsRasterBlock::subRect( const QgsRectangle &extent, int width, int height
   QgsDebugMsgLevel( QStringLiteral( "subRect: %1 %2 %3 %4" ).arg( subRect.x() ).arg( subRect.y() ).arg( subRect.width() ).arg( subRect.height() ), 4 );
   return subRect;
 }
+
+template<class T> bool findMaximum( const void *data, bool hasNoData, double noDataValue, int width, int height, double &maximum, std::size_t &offset )
+{
+  const T *castData = static_cast< const T * >( data );
+  auto end = castData + static_cast< std::size_t >( width ) * static_cast< std::size_t >( height );
+  if ( !hasNoData )
+  {
+    const auto maxElem = std::max_element( castData, end );
```

Comment:
you might run into trouble with float and double when they are NaN: https://stackoverflow.com/a/35890119/23076275 . Same for std::min_element and presumably minmax_element

Reply:
Eww, that's painful! I can't think of any alternative but to fallback to the non-std branch for double/float types then...

Reply:
Create a specific branch when ``if constexpr (std::is_same<T, float>::value || std::is_same<T, double>::value)`` that uses a custom NaN friendly comparison function